src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RasterSymbolizerBuilder.java

```diff
@@ -293,19 +326,19 @@ private ColorMap getColorMapFromBreaks(
                     if (i == 1) {
                         entry.setLabel(
                                 "< "
-                                        + format.format(value)
+                                        + getLabelForRaster(format, value)
                                         + getPercentagesLabelPortion(percentages, i - 1));
                     } else if (i == breaks.length - 1) {
                         entry.setLabel(
                                 ">= "
-                                        + format.format(prev)
+                                        + getLabelForRaster(format, prev)
                                         + getPercentagesLabelPortion(percentages, i - 1));
                     } else {
                         entry.setLabel(
                                 ">= "
-                                        + format.format(prev)
+                                        + getLabelForRaster(format, prev)
                                         + " AND < "
-                                        + format.format(value)
+                                        + getLabelForRaster(format, value)
```

Comment:
So this means we can now get a legend looking like ">= 10 AND < Infinity"?
That's ugly, if the high value is infinity, it would read better if it was just ">= 10"